### PR TITLE
Install composer if outdated

### PR DESCRIPTION
**Pull request by Ernestopheles**
I added a check in vendors.php: If composer already installed but versiion < 2.0.0., than install composer newly.

I need this for Contao CMS as composer >= 2.0.0 is required.

### CODE REVIEW

**Comment by Schrank** [comment]
Sounds reasonable to me. But can you please NOT delete composer.lock? Thanks :-)

**Comment by Ernestopheles** [comment]
Sorry, this was not intended. Would you, please, be so kind to fix that for me?

File: recipe/deploy/vendors.php [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-// composer version to `.dep/composer.phar`. To use specific composer version
-// download desired phar and place it at `.dep/composer.phar`.
+// If Composer is installed and version >= 2.0.0.: Returns Composer binary path in found.
+// Otherwise try to install latest composer version to `.dep/composer.phar`.
+// To use specific composer version, download desired phar and place it at `.dep/composer.phar`.
 set('bin/composer', function () {
     if (test('[ -f {{deploy_path}}/.dep/composer.phar ]')) {
-        return '{{bin/php}} {{deploy_path}}/.dep/composer.phar';
+        $composerVersion = run('{{bin/php}} {{deploy_path}}/.dep/composer.phar --version');
+        if (version_compare($composerVersion, '2.0.0', '>=')) {
```
Comment (antonmedv) [inline recipe/deploy/vendors.php:15]: Not sure if it is a good solution. Let's keep the version what user installed by default.

**Review by antonmedv — CHANGES_REQUESTED** [review]

**Comment by Ernestopheles** [comment]
Most often, I assume, it's the Hoster who not updated Composer. But anyway, may be some users want to keep an outdated version. So I agree to close this PR.

I can as good implement my own task to update composer.

**Pull request closed without merging**